checks.d/f5.py

```diff
@@ -0,0 +1,474 @@
+                    ('ltmServerSslStatConns',               'gauge'),
+                    ('ltmServerSslStatCachedCerts',         'gauge')
+                ]
+            ]
+
+
+    ]
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        """Initialize the class"""
+        if len(instances) < 1:                                                  # If there is less than one instance (Should never occur)
+            sys.exit(1)                                                         # Exit the program with an exit code of 1
+        os.environ["MIBDIRS"] = str(self.MIB_DIR)
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)    # Otherwise, call the super()
+
+    def get_session(self, host, user, authPassword, privPassword, port=161):
+        """Establish a session with the target F5, and return it"""
+        self.log.info("Establish session with "+str(host))          # log_level:info
+        try:                                                        # Try to
+            session = Session(                                      # Establish a session
+                                hostname=host,                      # Use the hostname from the instance
+                                version=3,                          # Default version is SNMP v3, other versions are insecure
+                                remote_port=port,                   # Use the specified port, defaulting to 161
+                                timeout=2,                          # Define the number of seconds before a timeout to be 2
+                                retries=3,                          # Define the maximum number of retries to be 3
+                                security_level="auth_with_privacy", # Use authentication with privacy
+                                security_username=user,             # Use the username from the instance
+                                privacy_protocol="AES",             # Define the privacy protocol to be AES
+                                privacy_password=privPassword,      # Use the privacy password from the instance
+                                auth_protocol="SHA",                # Define the authentication protocol to be SHA
+                                auth_password=authPassword          # Use the authentication password from the instance
+                            )                                       #
+            return session                                          # Otherwise, return the established snmp session with the F5
+        except Exception as e:                                      # If there are any exceptions
+            self.log.error("Could not establish session")           # log_level:error
+            self.log.error(e)                                       # log_level:error
+            sys.exit(1)                                             # Do not continue
```

Comment:
We don't want to kill the agent if you can't establish an exception. The agent will be restarted by the supervisor, so this will lead to some nasty side effects. Please raise an exception instead, it will stop the check (you should also clean up the logging, you should look at the other checks to see how we handle errors like this).